src/Stratis.Bitcoin.Features.Interop/CirrusContractClient.cs

```diff
@@ -110,7 +110,7 @@ public async Task<MultisigTransactionIdentifiers> MintAsync(string contractAddre
                 accountBytesPadded[0] = 9; // 9 = Address
                 Array.Copy(accountBytes, 0, accountBytesPadded, 1, accountBytes.Length);
 
-                byte[] amountBytes = this.serializer.Serialize((UInt256)amount.Satoshi);
+                byte[] amountBytes = this.serializer.Serialize(new UInt256(amount.ToByteArray()));
```

Comment:
Note: the SmartContracts UInt256 actually uses a BigInteger internally, so we don't have the same array padding issues here that the NBitcoin uint256 does elsewhere